### PR TITLE
Change registering order and force to send the cap parameter

### PR DESCRIPTION
**Pull request by eruizgar91**
## Description

Add a description of your changes here.

## Is this PR related with an open issue?

Related to Issue #

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 
## Checklist:

- [ ] Follows the code style of this project.
- [ ] Tests Cover Changes
- [ ] Documentation

#### Funny gif

Put a link of a funny gif inside the parenthesis-->

### CODE REVIEW

File: src/nevermined/Assets.ts [hunk 1]
```diff
@@ -375,7 +375,7 @@ export class Assets extends Instantiable {
         publisher: Account,
         assetRewards: AssetRewards = new AssetRewards(),
         method: string = 'PSK-RSA',
-        cap?: number,
+        cap: number,
```
Comment (aaitor) [inline src/nevermined/Assets.ts:378]: maybe make `0` by default like in the `royalties` parameter

File: src/nevermined/Assets.ts [hunk 2]
```diff
@@ -607,7 +607,7 @@ export class Assets extends Instantiable {
...
```
Comment (aaitor) [inline src/nevermined/Assets.ts:610]: same here

File: src/nevermined/Assets.ts (hunk 1 above)
Comment (eruizgar91) [reply src/nevermined/Assets.ts:378]: Does make sense create an NFT with cap 0?

**Review by r-marques — APPROVED** [review]
lgtm

**Pull request merged**